java-core/pom.xml

```diff
@@ -39,10 +39,11 @@
       </dependency>
       <!-- Test dependencies -->
       <dependency>
-        <groupId>junit</groupId>
-        <artifactId>junit</artifactId>
-        <version>4.13.2</version>
-        <scope>test</scope>
+        <groupId>org.junit</groupId>
+        <artifactId>junit-bom</artifactId>
+        <version>${junit.version}</version>
+        <type>pom</type>
+        <scope>import</scope>
```

Comment:
Per @suztomo's suggestion, I don't think we want to import junit bom in java-core, as I think it would go into [shared-dependencies](https://github.com/googleapis/sdk-platform-java/blob/main/java-shared-dependencies/first-party-dependencies/pom.xml#L45-L51). So I think we should declare those three test dependencies individually and manage the version through the parent pom, is that correct @suztomo ?

Reply:
> I don't think we want to import junit bom in java-core, as I think it would go into [shared-dependencies](https://github.com/googleapis/sdk-platform-java/blob/main/java-shared-dependencies/first-party-dependencies/pom.xml#L45-L51)

java-core it not imported by the shared dependencies BOM.

Having JUnit in third-party-dependencies is a good place. Because third-party-dependencies BOM does not go to the Libraries BOM.

Reply:
> java-core it not imported by the shared dependencies BOM.

I think it is included in the [first-party-dependencies](https://github.com/googleapis/sdk-platform-java/blob/main/java-shared-dependencies/first-party-dependencies/pom.xml#L45-L51), am I missing something?

Reply:
java-core imports shared dependencies in [parant pom](https://github.com/googleapis/sdk-platform-java/blob/d3d6af29aaab425fe3e7e72ef6899eb3b24df599/java-core/pom.xml#L35), why this doesn't cause a circular dependency?

Reply:
> am I missing something?

Importing a BOM and declaring dependencies are different. For a BOM, only imported BOMs matters.

Reply:
IIUC, first-party-dependencies imports google-cloud-core bom, which also imports junit bom, so junit bom would ends up in libraries bom?

Reply:
> IIUC, first-party-dependencies imports google-cloud-core bom, which also imports junit bom, so junit bom would ends up in libraries bom?

Actually, junit bom will not end up in libraries bom since only dependencies declared in google-clour-core will end up in libraries bom. junit, however, is a transitive dependency.

Reply:
> So I think we should declare those three test dependencies individually and manage the version through the parent pom

I think use junit-bom is better because `junit-platform-launcher` has a different version than `junit-jupiter-engine` and `junit-vintage-engine`.